2.6.0/Dockerfile

```diff
@@ -47,6 +45,11 @@ ADD yarn-site.xml $HADOOP_CONF_DIR/yarn-site.xml
 ADD mapred-site.xml $HADOOP_CONF_DIR/mapred-site.xml
 ADD hdfs-site.xml $HADOOP_CONF_DIR/hdfs-site.xml
 
+# Creating symlink for HADOOP configuration files
+RUN mkdir /data && \
+hdfs namenode -format
+VOLUME /data
```

Comment:
Is it correct to say that HDFS data is splitted in two locations:
1. The image layer contains the formatted data generated by the `hdfs namenode -format` command.
2. The container layer will contain any new data written to HDFS when using this hadoop image (using copy-on-write).

Assume the following scenario: we've been running a Hadoop cluster with the docker image for while and wish to upgrade Hadoop to the new version, while keeping our data. We change the Dockerfile and produce a new image version. The new image won't share the same formatted layer since the previous layers will be invalidated by the new hadoop version. We delete the current NameNode container and try to start the new NameNode. If Hadoop has changed its internal data structure for HDFS data, it is possible that the image layer's format be inconsistent with the volume layer's format.

It seems to me that we can avoid this problem but putting the VOLUME instruction before doing the format, but I may be missing something here...